src/task/TaskController.ts

```diff
@@ -51,11 +55,25 @@ export class TaskController extends Controller {
             return validationError.message;
         }
 
-        return Task.create({
-            ...task,
+        const folder = await Folder.findOneBy({ id: folderId });
+
+        if (!folder) {
+            this.setStatus(404);
+            return "Folder not found";
+        }
+
+        const folderIndex = await Task.count({
+            where: { folderId }
+        });
+
+        const task = await Task.create({
+            ...taskData,
+            folderIndex,
+            status: "open",
             authorId: req.ctx.state.user.id,
-            status: "open"
         }).save();
+
+        await Folder.merge(folder, { tasks: [...folder.tasks, task] }).save();
```

Comment:
I think you can remove `...folder.tasks` as `.merge()` should not override the current items. 

The current implementation might result in the current items being duplicated